mapswipe_workers/mapswipe_workers/generate_stats/project_stats.py

```diff
@@ -0,0 +1,345 @@
+import os
+from psycopg2 import sql
+import pandas as pd
+import datetime
+from mapswipe_workers import auth
+from mapswipe_workers.definitions import logger
+from mapswipe_workers.definitions import DATA_PATH
+from mapswipe_workers.utils import geojson_functions
+
+
+def id_to_string(x):
+    try:
+        x = str(int(x))
+    except:
+        x = str(x)
+
+    return x
+
+
+def get_results_by_project_id(filename, project_id):
+    '''
+    Export results for the given project id.
+
+    Parameters
+    ----------
+    filename: str
+    project_id: str
+    '''
+
+    pg_db = auth.postgresDB()
+    sql_query = sql.SQL(
+        'COPY (SELECT * FROM results WHERE project_id = {}) TO STDOUT WITH CSV HEADER'
+    ).format(sql.Literal(project_id))
+
+    with open(filename, 'w') as f:
+        pg_db.copy_expert(sql_query, f)
+    del pg_db
+    logger.info(f'got results from postgres for {project_id}')
+
+    df = pd.read_csv(filename)
```

Comment:
I would try to refactor this code bit to load a DataFrame from a CSV query into a function. You have it in at least 3 different places, so you could do something like:
```python
def load_df_from_sql(query):
   # your code above, or better, using pandas.read_sql_query
   return df
```